gems/aws-sdk-core/spec/aws/ec2_metadata_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require_relative '../spec_helper'
+
+module Aws
+  describe EC2Metadata do
+    let(:client) { EC2Metadata.new }
+
+    let(:metadata_path) { '/latest/meta-data/foo/' }
+    let(:token_path) { '/latest/api/token' }
+
+    it 'fetches a token before getting metadata' do
+      stub_request(
+        :put, "http://169.254.169.254#{token_path}"
```

Comment:
The address is repeated a bunch, might be good to extract.

Could also create a few quick helper methods for things like `stub_get_token` since thats used in so many places.